okio-files/src/jsMain/kotlin/okio/FileSink.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import fs.writeSync
+import org.khronos.webgl.Uint8Array
+
+internal class FileSink(
+  private val fd: Number
+) : Sink {
+  private var closed = false
+
+  override fun write(source: Buffer, byteCount: Long) {
+    require(byteCount >= 0L) { "byteCount < 0: $byteCount" }
+    require(source.size >= byteCount) { "source.size=${source.size} < byteCount=$byteCount" }
+    check(!closed) { "closed" }
+
+    val data = Uint8Array(byteCount.toInt())
+    data.set(source.readByteArray().toTypedArray())
```

Comment:
Working across Kotlin ByteArrays and JavaScript Uint8Arrays is pretty awful. They’re almost identical in capability, but I couldn’t find efficient ways to move data between them.

Reply:
Maybe worth filing a YouTrack issue?

Reply:
Probably. I need to do some research to file the right issue so that it’s actionable and compatible.

Reply:
Out of my depth here, but anything to learn from kotlinx-io?

https://github.com/Kotlin/kotlinx-io/blob/master/core/jsMain/src/kotlinx/io/buffer/Buffer.kt
https://kotlinlang.org/api/latest/jvm/stdlib/org.khronos.webgl/-data-view/